### PR TITLE
fix(pose26): enable kpts_sigma generation during validation for RLE loss

### PR DESCRIPTION
**Pull request by Miaoge-Ge**
## Description
This PR fixes a bug where `RLE Loss` calculation fails or is incorrect during validation mode in `Pose26` models.

## Issue
Currently, in `ultralytics/nn/modules/head.py`, the `kpts_sigma` (required for RLE loss) is only generated when `self.training` is `True`.
```python
if self.training:  # Current logic
    preds["kpts_sigma"] = ...
```
This causes the loss calculation to lack necessary sigma values during validation (when `self.training` is `False`).

## Fix
Updated the condition to generate `kpts_sigma` during both training and validation (non-export) modes:
```python
if self.training or not self.export:  # Fixed logic
    preds["kpts_sigma"] = ...
```

## Checklist
- [x] Changes validated on local machine (YOLO26 validation now computes RLE loss correctly).

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fixes YOLO26 pose validation by generating `kpts_sigma` when needed (non-export), ensuring RLE loss has required inputs ✅

### 📊 Key Changes
- Updated `ultralytics/nn/modules/head.py` to compute and include `preds["kpts_sigma"]` not only during training, but also during validation when `not self.export`.
- Added an inline comment clarifying that `kpts_sigma` is required for RLE loss during validation.

### 🎯 Purpose & Impact
- Prevents validation-time failures or incorrect behavior when using RLE loss that depends on `kpts_sigma`.
- Keeps export behavior intact by avoiding unnecessary `kpts_sigma` generation in export mode.
- Improves consistency between training and validation outputs for pose models using sigma-based losses.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Miaoge-Ge, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message, and an Ultralytics engineer will assist shortly. To ensure a seamless integration of your work, please review the following checklist:

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/nn/modules/head.py [hunk 1]
```diff
@@ -725,7 +725,7 @@ def forward_head(
             bs = x[0].shape[0]  # batch size
             features = [pose_head[i](x[i]) for i in range(self.nl)]
             preds["kpts"] = torch.cat([kpts_head[i](features[i]).view(bs, self.nk, -1) for i in range(self.nl)], 2)
-            if self.training:
+            if self.training or not self.export:  # kpts_sigma required for RLE loss in validation
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/modules/head.py:728]: 💡 **MEDIUM**: `if self.training or not self.export` will also generate `kpts_sigma` for *all* non-export inference paths (e.g., `model.predict()`), not just validation. That adds extra compute/memory and changes the default inference outputs. If the intent is strictly “validation only” for RLE loss, consider gating this behind a validation-specific condition or a loss/need-based flag (so regular inference doesn’t pay the cost or return extra tensors).

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean, minimal change, but the new condition will compute `kpts_sigma` for all non-export inference, not only validation. If that’s unintended, add a more specific gate to avoid unnecessary overhead and output changes during standard inference.

💬 Posted 1 inline comment
